File: sem1_home/task1.py
numbers=[4,1,2,7,6,9]

def sort_list_imperative(numbers):
    for i in range(1, len(numbers)):
        number = numbers[i]
        pos = i
        while pos > 0 and numbers[pos - 1] < number:
            numbers[pos] = numbers[pos - 1]
            pos -= 1
        numbers[pos] = number
    pass
    return numbers

def sort_list_declarative(numbers):
    numbers.sort(reverse=True)
    pass
    return numbers


print(sort_list_imperative(numbers))

print(sort_list_declarative(numbers))

File: sem1_home/task1.js
let numbers = [1, 3, 57, 4, 2, 4];

function sort_imp(numbers) {
  for (let index = 1; index < numbers.length; index++) {
    let number = numbers[index];
    position = index;
    while (position > 0 && numbers[position - 1] < number) {
      numbers[position] = numbers[position - 1];
      position -= 1;
    }
    numbers[position] = number;
  }
  return numbers;
}

function sort_dec(numbers) {
  numbers.sort((a, b) => b - a);
  return numbers;
}

console.log(sort_imp(numbers));
console.log(sort_dec(numbers));
